### ISSUE TITLE
Train pre-trained model

### ISSUE THREAD

**Issue by mrekin** [ultralytics/ultralytics#1790]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi!
Thanks for great tool :)
I'm new in ML and question may be a bit stupid.

I have pre-trained model with two classes [classA, classB]
I have new data set with two classes [classB, classA]

I want to train  existing model with new dataset.

As I understood - the same classes have different id's (0 and 1) and I'll have incorrect results after trainig with command
`yolo detect train data=newdataset.yaml model=yolov8s.yaml epochs=100 imgsz=640 pretrained=pretrained.pt`

Do I need to update label files and newdataset.yaml to correct classes id's 
or
I need only proper(the same) classes names and command above will work fine?

Thanks for help :)

**Comment by glenn-jocher** [comment]
Hello @kiwiflan, thank you for using YOLOv8 and for your question. 

Your understanding is correct, and you can certainly use the existing pre-trained model on a new dataset. The important thing is to ensure that the class names in your new dataset match the class names in the pre-trained model's class list exactly. 

You can update the label files of your new dataset to ensure that the classes (e.g. "classA" and "classB") correspond to the correct class indices (e.g. 0 and 1) in the pre-trained model. Alternatively, you could rename the class names in the pre-trained model's class list to "classB" and "classA" to match the class names in your new dataset. 

As long as the class names and indices match, you can train your existing model with the new dataset using the command you provided:

`yolo detect train data=newdataset.yaml model=yolov8s.yaml epochs=100 imgsz=640 pretrained=pretrained.pt`

I hope this helps! Let me know if you have any further questions.

**Comment by mrekin** [comment]
@glenn-jocher Thanks for reply
But not fully clear: do I need update dataset or not:)
I preffer not update)
If it needed - is there any tool to do such update?

**Comment by bit-scientist** [comment]
@mrekin you do need to update, namely, change label files' indices accordingly if you want to match with your existing dataset. If you prefer not to update, the model won't prefer to work correctly :smile:
Read the above content from Glenn carefully, he outlined everything you might need.

**Comment by glenn-jocher** [comment]
@bit-scientist sure, I'll clarify. Yes, you do need to update your dataset's label files if the classes in your dataset have different class indices than the pre-trained model's class list. This is because the model uses the class indices to identify and classify objects in the image. If the class indices are mismatched, the model won't be able to correctly classify objects in your new dataset. 

You can update your label files either by changing the class indices in your label files to match the indices in the pre-trained model's class list, or you can modify the pre-trained model's class list to match the class indices in your new dataset. Unfortunately, there is no automatic tool to do this, but it is a straightforward process that requires only minimal changes to the label files or the pre-trained model's class list. 

I hope this clears things up! Let me know if you have any further questions.
